include/cvc5/c/cvc5.h

```diff
@@ -0,0 +1,3936 @@
+
+/**
+ * Determine if given sort a tuple sort.
+ * @param sort The sort.
+ * @return True if given sort is a tuple sort.
+ */
+bool cvc5_sort_is_tuple(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is a record sort.
+ * @warning This function is experimental and may change in future versions.
+ * @param sort The sort.
+ * @return True if the sort is a record sort.
+ */
+bool cvc5_sort_is_record(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is an array sort.
+ * @param sort The sort.
+ * @return True if the sort is an array sort.
+ */
+bool cvc5_sort_is_array(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is a finite field sort.
+ * @param sort The sort.
+ * @return True if the sort is a finite field sort.
+ */
+bool cvc5_sort_is_ff(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is a Set sort.
+ * @param sort The sort.
+ * @return True if the sort is a Set sort.
+ */
+bool cvc5_sort_is_set(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is a Bag sort.
+ * @param sort The sort.
+ * @return True if the sort is a Bag sort.
+ */
+bool cvc5_sort_is_bag(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is a Sequence sort.
+ * @param sort The sort.
+ * @return True if the sort is a Sequence sort.
+ */
+bool cvc5_sort_is_sequence(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is an abstract sort.
+ * @param sort The sort.
+ * @return True if the sort is a abstract sort.
+ *
+ * @warning This function is experimental and may change in future versions.
+ */
+bool cvc5_sort_is_abstract(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is an uninterpreted sort.
+ * @param sort The sort.
+ * @return True if given sort is an uninterpreted sort.
+ */
+bool cvc5_sort_is_uninterpreted_sort(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is an uninterpreted sort constructor.
+ *
+ * An uninterpreted sort constructor has arity > 0 and can be instantiated to
+ * construct uninterpreted sorts with given sort parameters.
+ *
+ * @param sort The sort.
+ * @return True if given sort is of sort constructor kind.
+ */
+bool cvc5_sort_is_uninterpreted_sort_constructor(Cvc5Sort sort);
+
+/**
+ * Determine if given sort is an instantiated (parametric datatype or
+ * uninterpreted sort constructor) sort.
+ *
+ * An instantiated sort is a sort that has been constructed from
+ * instantiating a sort with sort arguments (see cvc5_sort_instantiate().
+ *
+ * @param sort The sort.
+ * @return True if given sort is an instantiated sort.
+ */
+bool cvc5_sort_is_instantiated(Cvc5Sort sort);
+
+/**
+ * Get the associated uninterpreted sort constructor of an instantiated
+ * uninterpreted sort.
+ *
+ * @param sort The sort.
+ * @return The uninterpreted sort constructor sort.
+ */
+Cvc5Sort cvc5_sort_get_uninterpreted_sort_constructor(Cvc5Sort sort);
+
+/**
+ * Get the underlying datatype of a datatype sort.
+ * @param sort The sort.
+ * @return The underlying datatype of a datatype sort.
+ */
+Cvc5Datatype* cvc5_sort_get_datatype(Cvc5Sort sort);
```

Comment:
Here (and below), if you return a pointer, does it mean that the user owns the pointer?  This should be explicitly documented.

Reply:
See above, we have to decide what to do here. And yes, this is generally not documented yet. This is not an oversight, my plan is to add this when it is implemented (to avoid stale documentation in case we decide to change the handling from this draft to the implementation).

Reply:
SGTM